fifteen_min/src/isochrone.rs

```diff
@@ -113,16 +120,26 @@ impl Isochrone {
             return None;
         }
 
-        let req = PathRequest::between_buildings(
-            map,
-            self.start,
-            to,
-            match self.options {
-                Options::Walking(_) => PathConstraints::Pedestrian,
-                Options::Biking => PathConstraints::Bike,
-            },
-        )?;
-        map.pathfind(req).ok()
+        let constraints = match self.options {
+            Options::Walking(_) => PathConstraints::Pedestrian,
+            Options::Biking => PathConstraints::Bike,
+        };
+
+        let all_paths: Vec<Path> = self
+            .start
+            .iter()
+            .map(|b_id| {
+                map.pathfind(PathRequest::between_buildings(map, *b_id, to, constraints).unwrap())
+                    .ok()
+                    .unwrap()
+            })
+            .collect();
+
+        all_paths
+            .iter()
+            .map(|path| (path, path.total_length()))
```

Comment:
You can collapse these 3 using https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key

(Perl has a pattern called https://en.wikipedia.org/wiki/Schwartzian_transform related to this. If you're sorting a big list by some transformation of the elements, and the transformation isn't cheap to calculate, you do this trick of first mapping into pairs. There are less comparisons from finding the minimum of a list than sorting it, but the trick could still be relevant.)

Reply:
Thank you, I finally got the concept behind min_by_key, it took me a bit to realize it sorts by the function but still returns the actual data, cool!